File: Lab/Lab01/Lab01.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

//função inteira para analisar o total de maiusculas
int tot(char *palavra)
{
    // definir dados
    int cont = 0;
    char letra = 'a';
    // metodo iterativo
    for (int x = 0; x < strlen(palavra); x = x + 1)
    {
        // atribuir caractere à letra
        letra = palavra[x];
        // realizar o teste para verificar se a letra é maiusucla
        if ('A' <= letra && letra <= 'Z')
        {
            cont = cont + 1;
        }
    }
    // retornar
    return (cont);
}

int main()
{
    //definir dados
      int total = 0;
      char palavra [80];
    //leitura de string enquanto diferente de fim
      do{
        scanf("%s", palavra); getchar();
        if ( strcmp(palavra, "FIM")){
            total = tot(palavra);
            printf("O total de M e %i\n", total);
        }
      }while(strcmp(palavra, "FIM"));
    //retornar
      return 0;
}

File: Lab/Lab01/Lab01.java


import java.util.*;

public class Lab01 {
  // metodo que retorna total de maiusculos
  public static int total(String palavra) {
    // definir dados
    int total = 0;
    int x = 0;
    // metodo iterativo de ler palavra letra por letra
    for (x = 0; x < palavra.length(); x = x + 1) {
      char letra = palavra.charAt(x);
      if ('A' <= letra && letra <= 'Z') {
        total++;
      }
    }
    // retornar
    return total;
  }

  // função main
  public static void main(String[] args) {
    // definir dados
    String palavra = "";
    int totM = 0;
    Scanner scanner = new Scanner(System.in);
    // fazer metódo enquanto a palavra 'FIM' não for encontrada
    do {
      palavra = scanner.nextLine();
      if (!palavra.equals("FIM")){
        totM = total(palavra);
        System.out.println("O total de maiusculas é " + totM);
      }
    } while (!palavra.equals("FIM"));
    //lembrar de fechar scanner
      scanner.close();
  }
}
